src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/MetaUtil.java

```diff
@@ -122,7 +124,7 @@ private static String replacePackageSeparatorsWithDot(String name) {
     public static String internalNameToJava(String name, boolean qualified, boolean classForNameCompatible) {
         switch (name.charAt(0)) {
             case 'L': {
-                String result = replacePackageSeparatorsWithDot(name.substring(1, name.length() - 1));
+                String result = replacePackageAndHiddenSeparators(name.substring(1, name.length() - 1), '/', '.');
```

Comment:
Consider extracting constants `/` (as `PACKAGE_SEP_INTERNAL` perhaps)  and `.` (as `HIDDEN_SEP_INTERNAL`). You could then also create the inversely defined constants `PACKAGE_SEP_JAVA` and `HIDDEN_SEP_JAVA`. It would make the code slightly easier to read.

Reply:
:+1: 

Done!